### DISCUSSION TITLE
how to use gltfpack in mac

### DISCUSSION THREAD

**Discussion by hahhh11** [zeux/meshoptimizer#482] [Q&A]
<img width="567" alt="image" src="https://user-images.githubusercontent.com/29700518/189881304-0ada8099-ee58-4745-bcc4-d4db5d84a752.png">
these some error when I use gltfpack -tc cammand，and I download gltfpack-macos.zip，but I dont know how to use it

**Comment by zeux** [comment]
You should be able to simply run the executable inside gltfpack-macos.zip. Due to Apple's restriction on running downloaded executables you will probably need to launch it first using Ctrl+Click, see https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/mac

**Reply by hahhh11** [reply]
Thanks for the answer, but in fact when I use Ctrl+Click to open the gltfpack file, I get this error, the file may have been stored with a different text encoding, or it may not be a text file.
![image](https://user-images.githubusercontent.com/29700518/190293705-43095146-8b5c-4d93-a3be-eed7785cfc01.png)

**Reply by zeux** [reply]
Ah, you might be running into a problem where the binary doesn't have the executable flag. You can fix it from command-line like this (specify the actual path to gltfpack file):

```
chmod a+x ~/Downloads/gltfpack
```

I'll need to see if the file in the .zip archive can automatically get the right permission when using GitHub Actions.

**Reply by zeux** [reply]
If this is easier here's a one-liner that downloads, extracts and fixes file permissions for gltfpack; you'd still need to Ctrl+Click the executable once I believe, but after that you can simply launch it via command line.

```
curl -L https://github.com/zeux/meshoptimizer/releases/latest/download/gltfpack-macos.zip >gltfpack.zip && unzip -o gltfpack.zip && chmod a+x gltfpack
```

**Reply by hahhh11** [reply]
Thank you very much. It is useful to use the above code.
